mindsdb_native/libs/controllers/predictor.py

```diff
@@ -23,9 +23,9 @@ def _get_memory_optimizations(df):
 
     mem_usage_ratio = df_memory/total_memory
 
-    sample_for_analysis = True if mem_usage_ratio >= 0.05 else False
-    sample_for_training = True if mem_usage_ratio >= 0.15 else False
-    disable_lightwood_transform_cache = True if mem_usage_ratio >= 0.15 else False
+    sample_for_analysis = True if mem_usage_ratio >= 0.1 or len(df) > pow(10,4) else False
```

Comment:
replacing `len(df)` with `df.shape[0] * df.shape[1]` makes more sense

Reply:
@maximlopin if you can, please make this change, it should indeed workout better to take into account length + nr columns.

Also, realistically, the training sampling and disabling the cache should probably happen lightwood-side.

Reply:
https://github.com/mindsdb/mindsdb_native/pull/84